### DISCUSSION TITLE
send power consumption data to UDP

### DISCUSSION THREAD

**Discussion by goeran03** [RavenSystem/esp-homekit-devices#1777] [MEPLHAA Scripts]
I have tried to change my working Gosund JSON configuration as follows (see bold addon):
{
  "c":{"l":2,"b":[{"g":13,"t":5}]},
  "a":[
    {
      "t":2,
      "0":{"r":[{"g":15},{"g":0,"v":1}]},
      "1":{"r":[{"g":15,"v":1},{"g":0}]},
      "b":[{"g":13}],
      "es":[{
        "t":75,
        "dt":[5,4,12],
        "vf":0.13334,
        "cf":0.01295,
        "pf":1.5312,
        "y0":[{"v":245,"0":{"m":[[1]]}}],
        "y1":[{"v":15,"0":{"m":[[1]]}}],
        **"y2":[{
          "v":0,
          "r":1,
          "0":{"h":[{"h":"192.168.179.12","p":56333,"m":13,"c":"v=#HAA@0000"}]}
        }]**
      }]
    }
  ]
}

with less success. How can I display the power consumption values in the UDP string?
I just get the same string in the message with no value replacement.

**Comment by RavenSystem** [comment]
```json
"c":"v=#HAA@0002"
```

**Reply by goeran03** [reply]
Hi, I've tried this. No change. The resulting string has no substitutions. I get always #HAA@0002. I must do wrong something else.

**Reply by goeran03** [reply]
Is there any other precondition....
I would expect to get something like "v=xx.x" but i get "v=#HAA@0002".
same with other numbers for instance  #HAA@0000
Thanks.

**Reply by goeran03** [reply]
tried TCP connection as well, same...

**Reply by RavenSystem** [reply]
Sending using UDP is not implemented, and sending over TCP RAW has a bug. Then, it only works with HTTP PUT and HTTP POST. I will fix TCP RAW and I will try to add for UDP.

**Reply by goeran03** [reply]
Thanks for looking into it

**Comment by RavenSystem** [comment]
Try BETA version: https://github.com/RavenSystem/haabeta

**Reply by goeran03** [reply]
For me it looks like the system is permanently rebooting once I have inserted the JSON. That is what I see at the console.

**Reply by RavenSystem** [reply]
Update to beta 5.

**Comment by goeran03** [comment]
Hi José

Now it works. Great. Thanks a lot.

What do you think about having a substitution not only for sensor values but for a device identifier as well. Something like the MAC-address or DNS-name or some like this.

This would help to separate incoming data on the receiver’s side and using same UDP-port.

 

Göran

 

 

 

Von: José Antonio Jiménez Campos ***@***.***> 
Gesendet: Dienstag, 8. März 2022 12:58
An: RavenSystem/esp-homekit-devices ***@***.***>
Cc: goeransaenger ***@***.***>; Author ***@***.***>
Betreff: Re: [RavenSystem/esp-homekit-devices] send power consumption data to UDP (Discussion #1777)

 

Update to beta 5.

**Reply by RavenSystem** [reply]
You can write what you want in content. If you want ro write any kind of ID, you can.

**Comment by goeran03** [comment]
What I mean is in case I have a lot of smart plugs for instance. So I could load the same JSON to each of them and they would send an unique identifier. 

 

Von: José Antonio Jiménez Campos ***@***.***> 
Gesendet: Dienstag, 8. März 2022 17:57
An: RavenSystem/esp-homekit-devices ***@***.***>
Cc: goeransaenger ***@***.***>; Author ***@***.***>
Betreff: Re: [RavenSystem/esp-homekit-devices] send power consumption data to UDP (Discussion #1777)

 

You can write what you want in content. If you want ro write any kind of ID, you can.
